libvast/src/system/indexer.cpp

```diff
@@ -30,15 +30,11 @@ namespace vast::system {
 
 active_indexer_actor::behavior_type
 active_indexer(active_indexer_actor::stateful_pointer<indexer_state> self,
-               const std::string& name, value_index_ptr value_index) {
+               const std::string& name, value_index_ptr index) {
+  VAST_ASSERT_CHEAP(!name.empty());
+  VAST_ASSERT_CHEAP(index);
```

Comment:
I think it would be good to ensure this in the type system somehow, in an earlier comment you mentioned the idea of a `active_indexer::verified_spawn_args` class, or if that is too much effort for one PR we could go with a `nonnull<value_index_ptr>`.

Reply:
I did not do this because of the extra code it requires, but I probably should because it's a cleaner abstraction. I'll push something in a bit.

Reply:
After not having the time to get to this for four days, I propose that we postpone this refactoring for a later time. What we've got here solves the problem at hand. I just don't have the bandwidth currently to work on refactorings like this.

Reply:
@lava ping